ee/api/test/test_organization.py

```diff
@@ -29,51 +29,96 @@ def test_delete_second_managed_organization(self):
-                "detail": f"Cannot remove organization since that would leave member {self.CONFIG_USER_EMAIL} organization-less, which is not supported yet.",
-                "code": "invalid_input",
-                "type": "validation_error",
-            },
-        )
-        self.assertEqual(response.status_code, 400)
-        self.assertTrue(Organization.objects.filter(id=org_id).exists())
 
-    def test_no_delete_organization_not_administrating(self):
-        organization, organization_membership, team = Organization.objects.bootstrap(self.user)
-        organization_membership = cast(OrganizationMembership, organization_membership)
-        organization_membership.level = OrganizationMembership.Level.MEMBER
-        organization_membership.save()
-        self.assertTrue(Organization.objects.filter(id=organization.id).exists())
-        self.assertTrue(Team.objects.filter(id=team.id).exists())
-        response = self.client.delete(f"/api/organizations/{organization.id}")
-        self.assertEqual(response.status_code, 403)
-        self.assertTrue(Organization.objects.filter(id=organization.id).exists())
-        self.assertTrue(Team.objects.filter(id=team.id).exists())
+        self.assertEqual(response.status_code, 204, "Did not successfully delete last organization on the instance")
+        self.assertFalse(Organization.objects.filter(id=org_id).exists())
+        self.assertFalse(Organization.objects.exists())
+
+        response_bis = self.client.delete(f"/api/organizations/{org_id}")
+
+        self.assertEqual(response_bis.status_code, 404, "Did not return a 404 on trying to delete a nonexistent org")
+
+    def test_no_delete_organization_not_owning(self):
+        for level in OrganizationMembership.Level:
+            self.organization_membership.level = level
+            self.organization_membership.save()
+            response = self.client.delete(f"/api/organizations/{self.organization.id}")
+            if level != OrganizationMembership.Level.OWNER:
+                potential_err_message = f"Somehow managed to delete the org as a level {level} (which is not owner)"
+                self.assertEqual(
+                    response.data,
+                    {
+                        "attr": None,
+                        "detail": "Your organization access level is insufficient.",
+                        "code": "permission_denied",
+                        "type": "authentication_error",
+                    },
+                    potential_err_message,
+                )
+                self.assertEqual(response.status_code, 403, potential_err_message)
+                self.assertTrue(self.organization.name, self.CONFIG_ORGANIZATION_NAME)
+            else:
+                potential_err_message = f"Somehow did not delete the org as a level {level} (which is owner)"
+                self.assertEqual(response.status_code, 204, potential_err_message)
+                self.assertTrue(self.organization.name, "Woof")
```

Comment:
If the organization is deleted, it shouldn't exist. I think you need to assert that the object does not exist in the DB (filtering by `pk`). As an alternative you can `assertRaises` calling `object.refresh_from_db()`